sklearn/model_selection/tests/test_search.py

```diff
@@ -448,16 +449,18 @@ def test_grid_search_bad_param_grid():
         " Single values need to be wrapped in a list"
         " with one element."
     )
+    search = GridSearchCV(clf, param_dict)
     with pytest.raises(ValueError, match=error_msg):
```

Comment:
I think this needs to be updated expect a `TypeError` with a new expected error message. Launch the test locally to see the failure of this specific test more quickly on your local machine before pushing:

```
pytest sklearn/model_selection/tests/test_search.py -k test_grid_search_bad_param_grid
```